### PR TITLE
Enable nullability in LinkArea and LinkAreaConverter

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in LinkArea and LinkAreaConverter.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7103)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/LinkArea.LinkAreaConverter.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                    char sep = culture.TextInfo.ListSeparator[0];
+                    string[] tokens = text.Split(new char[] { sep });
+                    int[] values = new int[tokens.Length];
+                    TypeConverter intConverter = TypeDescriptor.GetConverter(typeof(int));
+                    for (int i = 0; i < values.Length; i++)
+                    {
+                        values[i] = (int)intConverter.ConvertFromString(context, culture, tokens[i])!;
+                    }
+
+                    if (values.Length != 2)
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/LinkArea.LinkAreaConverter.cs:76]: you may want to move this to line 70.

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/LinkArea.LinkAreaConverter.cs]: I applied the suggestion on all three cases that I found.

**Comment by gpetrou** [comment]
@dreddy-work any more feedback on this or we could merge it?

**Review by RussKie — APPROVED** [review]

**Comment by gpetrou** [comment]
@dreddy-work friendly reminder about this one.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**